frontend/dashboard/components/RepoList/RepoList.tsx

```diff
@@ -1,311 +1,153 @@
-  );
-};
-
 export const RepoList = ({
-  repos = defaultArray,
+  repos = [],
   isLoading,
-  pageSize = DATAGRID_DEFAULT_PAGE_SIZE,
   isServerSort = false,
-  rowCount,
+  totalRows,
+  pageNumber,
+  pageSize = DATAGRID_DEFAULT_PAGE_SIZE,
+  pageSizeOptions = DATAGRID_PAGE_SIZE_OPTIONS,
   onPageChange,
-  onSortModelChange,
   onPageSizeChange,
-  pageSizeOptions = DATAGRID_PAGE_SIZE_OPTIONS,
-  sortModel,
-  disableVirtualization = false,
-}: IRepoListProps) => {
-  const [paginationModel, setPaginationModel] = React.useState<GridPaginationModel>({
-    pageSize,
-    page: 0,
-  });
-  const [modalOpen, setModalOpen] = useState<boolean>(false);
-
-  const handlePaginationModelChange = (newPaginationModel: GridPaginationModel) => {
-    if (newPaginationModel.page !== paginationModel.page) {
-      onPageChange?.(newPaginationModel.page);
-    }
-    if (newPaginationModel.pageSize !== paginationModel.pageSize) {
-      onPageSizeChange?.(newPaginationModel.pageSize as DATAGRID_PAGE_SIZE_TYPE);
-    }
-    setPaginationModel(newPaginationModel);
-  };
-
-  const [copyCurrentRepoName, setCopyCurrentRepoName] = useState('');
-
-  const { mutate: setStarredRepo } = useSetStarredRepoMutation();
-  const { mutate: unsetStarredRepo } = useUnsetStarredRepoMutation();
-  const copyModalAnchorRef = useRef(null);
+  onSortClick,
+}: RepoListProps): React.ReactElement => {
   const { t } = useTranslation();
+  const tableSize = 'small';
+
+  const columns: Columns = [
+    {
+      accessor: 'favoriteIcon',
+      heading: t('dashboard.favorite_status'),
+      sortable: false,
+      headerCellClass: classes.favoriteIconHeaderCell,
+    },
+    {
+      accessor: 'name',
+      heading: t('dashboard.name'),
+      sortable: true,
+      headerCellClass: classes.nameHeaderCell,
+      bodyCellFormatter: (repoFullName: string) => <RepoNameWithLink repoFullName={repoFullName} />,
+    },
+    {
+      accessor: 'createdBy',
+      heading: t('dashboard.created_by'),
+      sortable: true,
+      headerCellClass: classes.createdByHeaderCell,
+    },
+    {
+      accessor: 'updated',
+      heading: t('dashboard.last_modified'),
+      sortable: true,
+      headerCellClass: classes.lastUpdatedHeaderCell,
+      bodyCellFormatter: (date: string) =>
+        new Date(date).toLocaleDateString('nb', { dateStyle: 'short' }),
+    },
+    {
+      accessor: 'description',
+      heading: t('dashboard.description'),
+      sortable: true,
+    },
+    {
+      accessor: 'actionIcons',
+      heading: t('general.actions'),
+      sortable: false,
+      headerCellClass: classes.actionIconsHeaderCell,
+    },
+  ];
+
+  // The local table can sort all columns, but Gitea API does not support sorting by createdBy or description
+  const nonSortableAccessors = ['createdBy', 'description'];
+  const remotePaginationColumns = columns.map((column) => ({
+    ...column,
+    sortable: nonSortableAccessors.includes(column.accessor) ? false : column.sortable,
```

Comment:
Why not set the sortable for `createdBy` and `description` to false directly in the columns const?